### DISCUSSION TITLE
Ability to customize modals.

### DISCUSSION THREAD

**Discussion by qqqllppp** [rainbow-me/rainbowkit#979] [Ideas]
For the Connect/Account/Chain Modals, there is currently no way to customize them. This would be a nice feature.

For example for Chain Modal, it would be nice to have the ability to customize what chains to show, which may not be all the supported chains. This is useful where Eth mainnet chain is required to get ENS name but we don't necessarily want our users to use Eth mainnet.

**Comment by KholdStare** [comment]
Also interested in this. It would be helpful to put a message/banner specific to the app when the connect modal pops up.

**Reply by DanielSinclair** [reply]
@KholdStare Can you provide a little more detail of what you think that should look like? Do you mean customizing the Connect Modal, or displaying a banner within your dApp? We do offer a `Disclaimer` API that allows you to message a component to the Connect Modal. Zora leverages this API. https://www.rainbowkit.com/docs/custom-app-info#add-a-disclaimer
<img width="533" alt="Screenshot 2023-04-10 at 2 46 53 PM" src="https://user-images.githubusercontent.com/4412473/230971405-52de6db4-43f3-4f60-a767-5a5cac819fe9.png">

**Reply by KholdStare** [reply]
@DanielSinclair I was thinking about customizing the "What is a wallet" space on the wider modal. A component we can insert with a banner or message to tie things more closely with our dApp. Both from a branding perspective, as well as making it more intuitive for users new to crypto how it relates to the dApp itself.

**Reply by DanielSinclair** [reply]
@KholdStare It's something we have thought about, but we would like to avoid customization of most text so that it's consistent across dApps and will be translated when we rollout i18n. It sounds like you may actually want to create UI or a modal in front of RainbowKit to tell users why they're connecting a wallet. RainbowKit would be compatible with that approach; you don't need to keep the `ConnectButton` in the top right, and could use the `Custom ConnectButton` API to control your UI before calling `openConnectModal`. https://www.rainbowkit.com/docs/custom-connect-button

**Reply by KholdStare** [reply]
@DanielSinclair Thanks for the explanation. Will let you know if we have any other questions.

**Comment by kevinjoshi46b** [comment]
Rainbowkit devs I am interested in knowing if this discussion/issue is currently added to the roadmap?
Since, the example explained by @junhuang-ho is the exact use case that I was looking for and it would be really convenient if its directly added to rainbowkit itself.
Maybe I could even help building it 🙌

**Reply by DanielSinclair** [reply]
@kevinjoshi46b We actually do have a PR open for that! Hope to merge soon #1156

**Reply by ajaynarang** [reply]
@DanielSinclair  So, no plan to allow customizing the "What is a wallet" section on connect modal? I understand your point/concern; however, I believe letting users customize would still be a good feature.

**Reply by roto93** [reply]
It would be great if customization of Account modal is available. Or at least some way to overwrite the address part. For me, I need to show the user's domain name for other chain rather than Ethereum.
